### PR TITLE
Wh destination tests

### PR DESCRIPTION
**Pull request by achettyiitr**
## Description of the change

> Motivation being adding timeouts: Since now we are running Warehouse configuration tests whenever the user creates/updates the destination, there are timeouts at the cp router which is 60 sec. So in order to by pass these cp router timeouts we are adding timeouts during the connection step because of which we are adding timeouts. Also this includes some bugs fixes for warehouse configuration testing

## Notion Link

> https://www.notion.so/rudderstacks/Warehouse-configuration-testing-88b82b55f1af4fefb7b8ace9d1f13de3

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/mssql/mssql.go [hunk 1]
```diff
@@ -112,6 +114,10 @@ var partitionKeyMap = map[string]string{
 }
 
 func Connect(cred CredentialsT) (*sql.DB, error) {
+	return connectWithTImeout(cred, connectTimeout)
+}
+
+func connectWithTImeout(cred CredentialsT, connectionTimeout int) (*sql.DB, error) {
```
Comment (abhimanyubabbar) [inline warehouse/mssql/mssql.go:120]: Spelling Mistake: `connectWithTimeout`

Comment (achettyiitr) [reply warehouse/mssql/mssql.go:120]: Done.

File: warehouse/mssql/mssql.go [hunk 2]
```diff
@@ -660,14 +669,12 @@ func (ms *HandleT) TestConnection(warehouse warehouseutils.WarehouseT) (err erro
 		misc.IsConfiguredToUseRudderObjectStorage(ms.Warehouse.Destination.Config),
 	)
 
-	ms.Db, err = Connect(ms.getConnectionCredentials())
+	ms.Db, err = connectWithTImeout(ms.getConnectionCredentials(), int(timeOut/time.Second))
```
[the remark this answers is not in the document]
Comment (achettyiitr) [reply warehouse/mssql/mssql.go:672]: Done.

File: warehouse/azure-synapse/azure-synapse.go [hunk 3]
```diff
@@ -659,15 +667,14 @@ func (as *HandleT) AlterColumn(tableName string, columnName string, columnType s
 }
 
 func (as *HandleT) TestConnection(warehouse warehouseutils.WarehouseT) (err error) {
+	timeOut := 15 * time.Second
```
Comment (achettyiitr) [inline warehouse/azure-synapse/azure-synapse.go:670]: Test connection timeout.

**Review by Jayachand — APPROVED** [review]
LGTM

**Review by cisse21 — APPROVED** [review]

**Pull request merged**